vital/plugin_loader/plugin_loader_filter.h

```diff
@@ -0,0 +1,123 @@
+/*ckwg +29
+ * Copyright 2019 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef KWIVER_VITAL_PLUGIN_LOADER_FILTER_H
+#define KWIVER_VITAL_PLUGIN_LOADER_FILTER_H
+
+#include <kwiversys/DynamicLoader.hxx>
+
+#include <vital/vital_types.h>
+
+#include <memory>
+
+namespace kwiver {
+namespace vital {
+
+// base class of factory hierarchy
+class plugin_factory;
+class plugin_loader;
+
+using plugin_factory_handle_t = std::shared_ptr< plugin_factory >;
+
+// -----------------------------------------------------------------
+/** Interface to plugin loader filters.
+ *
+ *
+ */
+class plugin_loader_filter
+{
+public:
+  using DL = kwiversys::DynamicLoader;
```

Comment:
This is less a comment on this PR but a thought in general (since it's probably done everywhere) ... should we be exposing kwiversys in the public link interface?